src/rotator_library/async_locks.py

```
@@ -0,0 +1,388 @@
+            else:
+                await waiter.wait()
+
+            # Acquired via release
+            async with self._lock:
+                self._stats["acquires"] += 1
+                self._stats["current_concurrent"] += 1
+                self._stats["peak_concurrent"] = max(
+                    self._stats["peak_concurrent"],
+                    self._stats["current_concurrent"]
+                )
+            return True
```

Ruff (0.15.4)

[warning] Consider moving this statement to an `else` block

(TRY300)